students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+# Description: This program will allow a user to input a donor and a donation.
+# The user will also be able to send out thank you notes to the donors.
+# Developer: Ryan Hamersky
+# Date: 05/21/2018
+# Rev: New
+
+# -----Data Section-----
+str_name = str()    # --> User's name
+int_user_input = int()  # --> User's option.
+dic_donors_sorted = {}  # --> Sorted dictionary of donors.
+dic_donors = {"William Gates, III": [10000.0, 1], "Mark Zuckerberg": [20000.0, 1],
+                   "Jeff Bezos": [7800.0,4, 2500, 2000, 3000, 300],"Paul Allen": [5000.0,1],
+                   "Mike Scott": [25.0,1]} # --> Dic that stores all donor's information
+str_salutation = ""   # --> Email salutation
+str_body = ""         # --> Email body
+str_valediction = ""  # --> Email valediction
+
+
+# -----Process Section-----
+# This class creates an donor object.
+class Donor:
+    initial_donation = 1  # --> Field is used as the initial donation is constant with all donor objects.
+
+    def __init__(self, donor, donation):  # --> Constructs attributes donor and donation.
+        self.donor = donor
+        self.donation = donation
+
+########### END OF DONOR OBJECT CLASS###############
+
+# Methods in the Menu class will execute code based off of the user's menu choice.
+
+
+class Menu:
+    @staticmethod
+    def menu_options():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns int data (1-4) which is used to determine what menu option will be executed.
+        '''
+        while True:
+            try:
+                int_choice = int(input(str_name + ", please select an option above. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verify that the user selected an option 1 through option 4.
+            else:
+                if 1 <= int_choice <= 5:
```

Comment:
You could be using a [dict as switch](https://uwpce-pythoncert.github.io/PythonCertDevel/modules/DictionaryAsSwitch.html#dict-as-switch) to map user input to dict keys and you wouldn't need this extra validation (you could just catch `Key_Error` on bad user input for any value that isn't a key in the dict)